sdks/java/core/build.gradle

```diff
@@ -117,6 +132,113 @@ task packageTests(type: Jar) {
   classifier = "tests"
 }
 
+class ValidatesRunnerConfig {
+  // Task name prefix
+  String testPrefix
+  // Runner name in documentation string
+  String runnerName
+  // List of test categories to exclude from this task. Optional.
+  List<String> excludes
+  // Pipeline options command line arguments.
+  // TODO: Can option keys be specified multiple times?
+  Map<String, String> pipelineOptions
+  // Configuration to use for test runtime classpath.
+  FileCollection configuration
+  // Additional system properties to be set for tests. Optional.
+  Map<String, String> systemProperties
+}
+
+def createValidatesRunner(Map m) {
+  def config = m as ValidatesRunnerConfig
+  assert config.testPrefix != null
+  assert config.runnerName != null
+  assert config.pipelineOptions != null
+  assert config.configuration != null
+  tasks.create(name: "${config.testPrefix}ValidatesRunner", type: Test) {
+    group = "Verification"
+    description = "Validate ${config.runnerName} runner"
+    def optionsList = config.pipelineOptions.collect {
+      // Escape strings with embedded '\' and '"' characters.
+      // TODO: Verify that this happens correctly between JSON and command line parsing.
+      def key = it.getKey().replaceAll("\\\\", "\\\\\\\\").replaceAll("\"", "\\\\\"")
+      def value = it.getValue().replaceAll("\\\\", "\\\\\\\\").replaceAll("\"", "\\\\\"")
+      "\"--${key}=${value}\""
+    }
+    def pipelineOptions = "[${optionsList.join(",")}]"
+
+    systemProperty "beamTestPipelineOptions", pipelineOptions
+    if (config.systemProperties != null) {
+      for (Map.Entry<String, String> property : config.systemProperties) {
+        systemProperty property.getKey(), property.getValue()
+      }
+    }
+    // TODO: Does Spark require a different forking strategy?
+    maxParallelForks 4
+    classpath = config.configuration
+    useJUnit {
+      includeCategories 'org.apache.beam.sdk.testing.ValidatesRunner'
+      if (config.excludes != null) {
```

Comment:
null -> false in Groovy

if (config.excludes) {

}

Reply:
Done.